agent/consul/auto_encrypt.go

```diff
@@ -18,7 +18,7 @@ const (
 	retryJitterWindow = 30 * time.Second
 )
 
-func (c *Client) RequestAutoEncryptCerts(servers []string, port int, token string, interruptCh chan struct{}) (*structs.SignedResponse, string, error) {
+func (c *Client) RequestAutoEncryptCerts(servers []string, port int, token string, interruptCh <-chan struct{}) (*structs.SignedResponse, string, error) {
```

Comment:
minor: Passing in `context.Context`as the first arg instead of the channel might be nice for future changes. I believe it'll require an additional 2 line change here (https://github.com/hashicorp/consul/pull/7609/files#diff-15078412ddffdf50ee505b3890dce6e4).

Reply:
Done.